kubeflow/core/kubeform_spawner.py

```diff
@@ -177,6 +177,26 @@ def _expand_user_properties(self, template):
 c.KubeSpawner.singleuser_working_dir = '/home/jovyan'
 volumes = []
 volume_mounts = []
+
+# Allow environment vars to override uid and gid.
+# This allows local host path mounts to be read/writable
+env_uid = os.environ.get('NOTEBOOK_UID')
```

Comment:
@abhi-g How does this compare to the current behavior? i.e. if you run on GKE with a PV for /home/jovyan how does this code changing the UID and GID change what user and group we run the notebook as?

We had a lot of trouble getting the permissions correct when using a PV for the home directory.
The thing to do would be to run it on GKE with the default settings and verify that the user had RW access to the home directory /home/jovyan

Reply:
I'll definitely test that. Except, last time I tried, I still get stuck with IAM issues on GKE